pkg/machinery/resources/k8s/static_pod_list_url.go

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package k8s
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/talos-systems/talos/pkg/machinery/proto"
+)
+
+// StaticPodListURLType is type of StaticPodListURL resource.
+const StaticPodListURLType = resource.Type("StaticPodListURLs.kubernetes.talos.dev")
```

Comment:
Feels like probably not a good name, `StaticPodServerStatus` ? 

if we have a static pod HTTP server, this provides a status of it, i.e. once the resource exists, server is running and the URL/endpoint of it is available here.